mongodbatlas/fw_provider.go

```diff
@@ -0,0 +1,394 @@
+					cstmvalidator.ValidDurationBetween(15, 12*60),
+					stringvalidator.ConflictsWith(path.Expressions{
+						path.MatchRelative().AtParent().AtName("duration_seconds"),
+					}...),
+				},
+			},
+			"duration_seconds": schema.Int64Attribute{
+				Optional:           true,
+				DeprecationMessage: "Use assume_role.duration instead",
+				Description:        "The duration, in seconds, of the role session.",
+				Validators: []validator.Int64{
+					int64validator.Between(900, 43200),
+					int64validator.ConflictsWith(path.Expressions{
+						path.MatchRelative().AtParent().AtName("duration"),
+					}...),
+				},
+			},
+			"external_id": schema.StringAttribute{
+				Optional:    true,
+				Description: "A unique identifier that might be required when you assume a role in another account.",
+				Validators: []validator.String{
+					stringvalidator.LengthBetween(2, 1224),
+					stringvalidator.RegexMatches(regexp.MustCompile(`[\w+=,.@:/\-]*`), ""),
+				},
+			},
+			"policy": schema.StringAttribute{
+				Optional:    true,
+				Description: "IAM Policy JSON describing further restricting permissions for the IAM Role being assumed.",
+				Validators: []validator.String{
+					cstmvalidator.StringIsJSON(),
+				},
+			},
+			"policy_arns": schema.SetAttribute{
+				ElementType: types.StringType,
+				Optional:    true,
+				Description: "Amazon Resource Names (ARNs) of IAM Policies describing further restricting permissions for the IAM Role being assumed.",
+			},
+			"role_arn": schema.StringAttribute{
+				Optional:    true,
+				Description: "Amazon Resource Name (ARN) of an IAM Role to assume prior to making API calls.",
+			},
+			"session_name": schema.StringAttribute{
+				Optional:    true,
+				Description: "An identifier for the assumed role session.",
+				Validators: []validator.String{
+					stringvalidator.LengthBetween(2, 64),
+					stringvalidator.RegexMatches(regexp.MustCompile(`[\w+=,.@\-]*`), ""),
+				},
+			},
+			"source_identity": schema.StringAttribute{
+				Optional:    true,
+				Description: "Source identity specified by the principal assuming the role.",
+				Validators: []validator.String{
+					stringvalidator.LengthBetween(2, 64),
+					stringvalidator.RegexMatches(regexp.MustCompile(`[\w+=,.@\-]*`), ""),
+				},
+			},
+			"tags": schema.MapAttribute{
+				ElementType: types.StringType,
+				Optional:    true,
+				Description: "Assume role session tags.",
+			},
+			"transitive_tag_keys": schema.SetAttribute{
+				ElementType: types.StringType,
+				Optional:    true,
+				Description: "Assume role session tag keys to pass to any subsequent sessions.",
+			},
+		},
+	},
+}
+
+func (p *MongodbtlasProvider) Configure(ctx context.Context, req provider.ConfigureRequest, resp *provider.ConfigureResponse) {
+	var (
+		data tfMongodbAtlasProviderModel
+	)
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	defaultValuesAndValidations(&data, resp)
```

Comment:
this is really hard to read. instead of using a pointer, let's return the object as part of the function

Reply:
it is okay to pass a pointer as input. IMO what decreases readability is to change the value of the object of the pointer inside the function. A better approach is to get the pointer as input and then return a new object 
Example`data := defaultValuesAndValidations(&data, resp)`

Reply:
correct, this is aligned with my most recent commit